src/matrix/matrix-lib-test.cc

```diff
@@ -4740,10 +4740,11 @@ template<typename Real> static void MatrixUnitTest(bool full_test) {
 
 int main() {
   using namespace kaldi;
+  static_assert(int(kNoTrans) != int(CblasNoTrans) && int(kTrans) == int(CblasTrans), 
```

Comment:
Don't put this here. I recommend putting it into one of the non-test .cc files. By putting it into the test files, you are preventing the static_assert from being run when someone just types "make" instead of "make test".  There's no good "general" .cc file I can think of, so I recommend putting it at the top of kaldi-matrix.cc.

Also, I'm not positive, but can you do `kNoTrans == CblasNoTrans` here? Casting to int should be fine since the literal values will fit into an int, but it's actually not defined in C/C++ what kind of int backs an enum, so you should prefer to avoid the cast if possible.

Also, you can put the static_assert into global scope. It does not need to sit inside a function.